frontend/src/components/cart-page/CartSummarySection.tsx

```diff
@@ -0,0 +1,66 @@
+import { useContext } from 'react'
+import { useNavigate } from 'react-router-dom';
+
+import { ShoppingCartContext } from '../../contexts/shopping-cart-context/shoppingCartContext';
+import { Box, Button, Card, CardContent, Divider, Typography } from "@mui/material";
+
+import cartStyles from './cartStyles';
+const {
+    totalCardStyle,
+    buttonContainerStyle,
+    buttonStyle,
+} = cartStyles;
+
+const CartSummarySection = () => {
+    const navigate = useNavigate();
+    const { cartItems, userSignin } = useContext(ShoppingCartContext);
+
+    const handleProceedToCheckout = () => {
+        userSignin
+            ? navigate("/shipping")
+            : navigate("/user/signin?redirect=/shipping");
+    };
+
+    const calculateTotal = () => {
+        const totalItems = cartItems.reduce((a, c) => a + c.quantity, 0);
+        const totalPrice = cartItems.reduce((a, c) => a + c.price * c.quantity, 0);
+        return `Total (${totalItems} items) : $${totalPrice}`;
+    };
```

Comment:
let's do 3 things here:
1- rename to calculateTotalPrice & calculateTotalItems
2- let it handle only the calculation and let the text be somewhere else.
3- wrap the functions with useMemo to prevent unnecessary re-renders = faster performance

Reply:
Thanx